### ISSUE TITLE
error should end iterator

### ISSUE THREAD

**Issue by niklasf** [DougLau/gift#11]
The iterator would be safer to use if it would end after an error (i.e. if next returns `None` at least once after an error).

Example that loops forever:

```rust
fn main() {
    for frame in gift::Decoder::new(std::io::Cursor::new(b"")).into_frames() {
        println!("{:?}", frame);
    }
}
```

```
Err(UnexpectedEndOfFile)
Err(UnexpectedEndOfFile)
Err(UnexpectedEndOfFile)
[...]
```

**Comment by DougLau** [comment]
Fixed in e7153b04270b6fec6cc6b3e718244f9a9105566e

**Comment by niklasf** [comment]
Awesome, thanks!
